test/augmentation/test_augmentation.py

```diff
@@ -25,6 +25,334 @@
 )
 
 
+from kornia.testing import BaseTester
+from kornia.augmentation.base import AugmentationBase2D
+
+# TOTEST
+# _validate_input_dtype
+# _transform_input
+# identity_matrix
+
+# Common
+# Given arbitrary input shapes the output dims are consistent
+# shape changing not shape changing
+#  return_transform: bool = False, same_on_batch: bool = False, p: float = 0.5,
+#  p_batch: float = 1.
+# Smoke tests, lehet példányosítani, meg lehet hívni az abstract függvényeit stb
+# generate_parameters annyi paramot ad vissza amennyi a batch size
+# compute_transformation implementálva van és batchsiye stimmel
+# apply_transform implementálva van és batchsiye stimmel
+
+
+class CommonTests(BaseTester):
+
+    ############################################################################################################
+    # Attributes variables to set
+    ############################################################################################################
+    _augmentation_cls = None
+    _default_param_set = None
+
+    ############################################################################################################
+    # Fixtures
+    ############################################################################################################
+    @pytest.fixture(scope="class")
+    def param_set(self, request):
+        raise NotImplementedError("param_set must be overriden in subclasses")
+
+    ############################################################################################################
+    # Test cases
+    ############################################################################################################
+    def test_smoke(self, device, dtype, param_set):
```

Comment:
I'm thinking that both `device` and `dtype` should be members of BaseTest class so that we can relieve each test call signature and just use `self.device` or `self.dtype` inside the implementation functions